### PR TITLE
BaSD-MS-49: Implement React Final Form in Applications and Interviews

### PR DESCRIPTION
**Pull request by PabloMBernal**
Task:
- Modify form to use react final form in applications and interviews component
- Add required function
- Add validate function
- Use submitting in Fields and buttons
- Use pristine in buttons

### CODE REVIEW

**Review by GuidoGlielmi — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -47,8 +47,8 @@
     ]
   },
   "engines": {
-    "node": "16.8.0",
-    "npm": "7.21.0"
+    "node": "14.x",
+    "npm": "8.x"
```
Comment (nicolasblasco) [inline package.json:51]: @PabloMBernal we shouldn't update any dependencies unless it was discussed with the team

File: src/Components/Admin/Applications/Form/index.js [hunk 2]
```diff
@@ -124,62 +106,74 @@ const Form = () => {
...
+                  disabled={isLoading}
+                />
+              </div>
+            </div>
+            <div className={styles.button}>
+              <ButtonCancel
+                disabled={isLoading}
+                onClick={() => history.push('/admin/applications')}
+              />
+              <ButtonConfirm type="submit" disabled={isLoading} />
```
Comment (nicolasblasco) [inline src/Components/Admin/Applications/Form/index.js:169]: @PabloMBernal we should change the disabled to have the formProps submitting and pristine

File: src/Components/Admin/Interviews/Form/index.js [hunk 3]
```diff
@@ -166,79 +150,98 @@ const Form = () => {
...
+                  disabled={formProps.submitting}
+                />
+              </div>
+            </div>
+            <div className={styles.button}>
+              <ButtonCancel
+                disabled={isLoading}
+                onClick={() => history.push('/admin/interviews')}
+              />
+              <ButtonConfirm type="submit" disabled={isLoading} />
```
Comment (nicolasblasco) [inline src/Components/Admin/Interviews/Form/index.js:237]: @PabloMBernal we should change the disabled to have the formProps submitting and pristine

File: src/Components/Admin/Interviews/Form/index.js [hunk 4]
```diff
@@ -166,79 +150,98 @@ const Form = () => {
     { _id: 'confirmed', value: 'confirmed', name: 'Confirmed' }
   ];
 
+  const required = (value) => (value ? undefined : 'Required');
+
```
Comment (nicolasblasco) [inline src/Components/Admin/Interviews/Form/index.js:154]: @PabloMBernal we should also create a validate function to validate the notes in the TextArea component

File: src/Components/Admin/Applications/Form/index.js [hunk 5]
```diff
@@ -124,62 +106,74 @@ const Form = () => {
     });
   };
 
+  const required = (value) => (value ? undefined : 'Required');
+
```
Comment (nicolasblasco) [inline src/Components/Admin/Applications/Form/index.js:110]: @PabloMBernal we should also create a validate function to validate the description in the TextArea component

**Review by nicolasblasco — CHANGES_REQUESTED** [review]
@PabloMBernal implmentation of final form is very good, but we should see the comments made and fix any changes requested

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**